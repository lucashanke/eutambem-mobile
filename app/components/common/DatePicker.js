import React from 'react';
import PropTypes from 'prop-types';
import ExternalDatePicker from 'react-native-datepicker';
import { StyleSheet } from 'react-native';

import appStyles, { DARK_PURPLE, GREY, PLACEHOLDER_GREY } from '../../styles';
import input from '../hoc/input';

export const style = StyleSheet.create({
  btnTextCancel: {
    color: GREY,
    fontSize: 17,
  },
  btnTextConfirm: {
    color: DARK_PURPLE,
    fontSize: 17,
  },
  placeholderText: {
    color: PLACEHOLDER_GREY,
  },
  date: {
    width: '100%',
    marginTop: 10,
    marginBottom: 10,
    height: 30,
    alignItems: 'flex-start',
    justifyContent: 'center',
  },
});

export const DatePicker = props => (
  <ExternalDatePicker
    {...props}
    onDateChange={props.onValueChange}
    date={props.value}
    mode="date"
    format="DD/MMM/YYYY"
    confirmBtnText="Ok"
    cancelBtnText="Cancelar"
    showIcon={false}
    style={style.date}
    customStyles={{
      dateInput: appStyles.input,
      placeholderText: style.placeholderText,
      btnTextCancel: style.btnTextCancel,
      btnTextConfirm: style.btnTextConfirm,
    }}
  />
);

DatePicker.propTypes = {
  value: PropTypes.string,
  onValueChange: PropTypes.func.isRequired,
}

DatePicker.defaultProps = {
  value: null,
};

export default input(DatePicker);
