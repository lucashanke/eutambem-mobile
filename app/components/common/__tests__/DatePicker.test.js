import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import ExternalDatePicker from 'react-native-datepicker';

import { DatePicker } from '../DatePicker';

const props = {
  onValueChange: () => {},
  value: '16/Oct/2018',
};

describe('DatePicker', () => {
  it('renders external datepicker component', () => {
    const wrapper = shallow(<DatePicker {...props} />);

    expect(wrapper.find(ExternalDatePicker)).toHaveLength(1);
  });

  it('renders passes value prop to external component date prop', () => {
    const wrapper = shallow(<DatePicker {...props} />);

    expect(wrapper.find(ExternalDatePicker).props().date).toEqual('16/Oct/2018');
  });

  it('calls onValueChange prop when date changes on external component', () => {
    const onValueChange = sinon.spy();
    const wrapper = shallow(<DatePicker
      {...props}
      onValueChange={onValueChange}
    />);

    wrapper
      .find(ExternalDatePicker)
      .props()
      .onDateChange('17/Oct/2018');

    expect(onValueChange.calledWith('17/Oct/2018')).toBeTruthy();
  });
});
