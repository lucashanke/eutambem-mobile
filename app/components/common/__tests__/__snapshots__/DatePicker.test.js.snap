// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders correctly 1`] = `
<ForwardRef(View)
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <ForwardRef(View)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <DatePicker
      onValueChange={[Function]}
      required={false}
      value={null}
    />
  </ForwardRef(View)>
</ForwardRef(View)>
`;
